src/xma/include/lib/xma_buffer.hpp

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (C) 2019 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef _XMA_BUFFERCLS_H_
+#define _XMA_BUFFERCLS_H_
+
+#include <cstdint>
+#include <cstring>
+#include <string>
+#include <atomic>
+#include "xrt.h"
+#include "core/include/experimental/xrt_bo.h"
+
+namespace xma_core {
+namespace plg {
+
+using memory_group = xrtMemoryGroup;
+
+/**
+ * class buffer : abstraction for xma buffer
+ */
+class buffer
+{
+private:
+    void*    dummy{nullptr};
+    uint8_t* data{nullptr};
```

Comment:
I don't know what dummy and data are?  But if it is the mapped host buffer part of bo, then this too is part of xrt::bo.

Reply:
changed. Though nothing wrong in having it here.